### PR TITLE
Ticket 14 - Completed

### PR DESCRIPTION
**Pull request by kieron-day**
Get Request - Articles By ID
Happy Path - 200 - responds with correct article object requested
Sad Path - 400 - invalid article id type (e.g. string)
Sad Path - 404 - path not found (already handled in previous ticket - Moved to global error handling describe block)
Sad Path 404 - article request is ok but does not exist in database
Sad Path 500 - Server Error

Creates error controller to tidy up app.js

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -11,10 +11,21 @@ beforeEach(() => {
 afterAll(() => db.end());
 
 describe("app", () => {
+	describe("Global Error Handling Tests", () => {
+		test("Status: 404, responds with 'Bad Request - Path Not Found'", () => {
```
Comment (cjplatten) [inline __tests__/app.test.js:15]: It's a bit fussy but 404 is just Not Found, Bad Request is for status 400! Check http.cat or http.dog if you're ever not fully sure!

File: controllers/error.controllers [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+};
+
+exports.errorPsql = (err, req, res, next) => {
+	if (err.code === "22P02") res.status(400).send({ message: "Bad Request" });
+	else next(err);
+};
+
+exports.error500s = (err, req, res, next) => {
+	res.status(500).send({ message: "Server Error" });
+};
```
Comment (cjplatten) [inline controllers/error.controllers]: This file isn't a .js file

**Review by cjplatten — COMMENTED** [review]
A couple of comments but otherwise looking good!

File: __tests__/app.test.js (hunk 1 above)
Comment (kieron-day) [reply __tests__/app.test.js:15]: Thanks!

File: controllers/error.controllers (hunk 2 above)
Comment (kieron-day) [reply controllers/error.controllers]: Not sure how that happened - thanks!

**Pull request merged**